src/monitor/pgautofailover--1.3--1.4.sql

```diff
@@ -0,0 +1,782 @@
+--
+-- extension update file from 1.3 to 1.4
+--
+-- complain if script is sourced in psql, rather than via CREATE EXTENSION
+\echo Use "CREATE EXTENSION pgautofailover" to load this file. \quit
+
+DROP FUNCTION IF EXISTS pgautofailover.register_node(text,text,int,name,int,
+                          pgautofailover.replication_state,text, int, bool);
+
+DROP FUNCTION IF EXISTS pgautofailover.node_active(text,text,int,int,int,
+                          pgautofailover.replication_state,bool,pg_lsn,text);
+
+DROP FUNCTION IF EXISTS pgautofailover.get_other_nodes
+                          (text,integer,pgautofailover.replication_state);
+
+DROP FUNCTION IF EXISTS pgautofailover.current_state(text);
+
+DROP FUNCTION IF EXISTS pgautofailover.current_state(text, int);
+
+ALTER TYPE pgautofailover.replication_state RENAME TO old_replication_state;
+
+CREATE TYPE pgautofailover.replication_state
+    AS ENUM
+(
+    'unknown',
+    'init',
+    'single',
+    'wait_primary',
+    'primary',
+    'draining',
+    'demote_timeout',
+    'demoted',
+    'catchingup',
+    'secondary',
+    'prepare_promotion',
+    'stop_replication',
+    'wait_standby',
+    'maintenance',
+    'join_primary',
+    'apply_settings',
+    'prepare_maintenance',
+    'wait_maintenance',
+    'report_lsn',
+    'fast_forward',
+    'join_secondary'
+ );
+
+-- Note the double cast here, first to text and only then to the new enums
+ALTER TABLE pgautofailover.node
+      ALTER COLUMN goalstate DROP NOT NULL,
+      ALTER COLUMN goalstate DROP DEFAULT,
+
+      ALTER COLUMN goalstate
+              TYPE pgautofailover.replication_state
+             USING goalstate::text::pgautofailover.replication_state,
+
+      ALTER COLUMN goalstate SET DEFAULT 'init',
+      ALTER COLUMN goalstate SET NOT NULL,
+
+      ALTER COLUMN reportedstate
+              TYPE pgautofailover.replication_state
+             USING reportedstate::text::pgautofailover.replication_state;
+
+ALTER TABLE pgautofailover.event
+      ALTER COLUMN goalstate
+              TYPE pgautofailover.replication_state
+             USING goalstate::text::pgautofailover.replication_state,
+
+      ALTER COLUMN reportedstate
+              TYPE pgautofailover.replication_state
+             USING reportedstate::text::pgautofailover.replication_state;
+
+DROP TYPE pgautofailover.old_replication_state;
+
+ALTER TABLE pgautofailover.formation
+      ALTER COLUMN number_sync_standbys
+       SET DEFAULT 0;
```

Comment:
I guess we should change all current formations to use `number_sync_standbys = 0`, right?

Reply:
Good point, done now!